modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectAction/fetchUtil.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClaySelect} from '@clayui/form';
+import {API} from '@liferay/object-js-components-web';
+
+export type ObjectsOptionsList = Array<
+	(
+		| React.ComponentProps<typeof ClaySelect.Option>
+		| React.ComponentProps<typeof ClaySelect.OptGroup>
+	) & {
+		options?: Array<React.ComponentProps<typeof ClaySelect.Option>>;
+		type?: 'group';
+	}
+>;
+
+export async function fetchObjectDefinitions(
+	objectDefinitionsRelationshipsURL: string,
+	values: Partial<ObjectAction>,
+	setRelationships: (values: ObjectDefinitionsRelationship[]) => void,
+	setObjectOptions: (values: ObjectsOptionsList) => void
+) {
+	const relationships = await API.fetchJSON<ObjectDefinitionsRelationship[]>(
+		objectDefinitionsRelationshipsURL
+	);
+
+	const relatedObjects: LabelValueObject[] = [];
```

Comment:
@danvb112 Please, if the declarations don't need to follow a specific order for dependency purposes, sort them alphabetically. Thanks :)